src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java

```diff
@@ -0,0 +1,15 @@
+/*
+IllegalInstantiation
+classes = java.util.Map, java.util.HashMap
+tokens = (default)CLASS_DEF
+
+
+*/
+
+//non-compiled syntax: the import is missing and is used to test the code
```

Comment:
It is our limitation, we do not operate on bad file at all.

Reply:
We keep only very bad files.
A bit wrong is not even considered.

Reply:
Can you forbid class that do not need import ?

Reply:
Test case removed.